src/Entities/Player.cpp

```diff
@@ -867,14 +866,16 @@ void cPlayer::SetFlyingMaxSpeed(double a_Speed)
 void cPlayer::SetCrouch(bool a_IsCrouched)
 {
 	// Set the crouch status, broadcast to all visible players
-
-	if (a_IsCrouched == m_IsCrouched)
+	if (!cRoot::Get()->GetPluginManager()->CallHookPlayerCrouching(*this, a_IsCrouched))
```

Comment:
If this returns false, should we tell the player that their crouch has been rejected and re-set their pose to normal? This needs testing.

Reply:
ah, indeed! i'll start working asap